versioned_docs/version-2020fa/finalproject.md

```diff
@@ -0,0 +1,141 @@
+---
+id: finalproject
+title: Final Project
+---
+
+This semester, we've learned some great tools for
+frontend, backend, and database development. Now,
+we will put them all together to make an amazing final
+project!
+
+## Outline
+
+This project will involve building a web app complete
+with a backend that connects to a Firestore database.
+
+You will be working in **groups of 2** to complete this final
+project. Your method of collaboration is up to you,
+but **using GitHub is strongly advised**. VS Code also has
+an excellent live coding tool which you can use if you prefer
+to do all work together.
+
+## Requirements
+
+- Firebase authentication system
+- Create a backend using GET and POST requests
+- Have a React frontend which talks to the backend and
+  puts/fetches data from the database.
+- Use conditional rendering (at least once)
+- Use ES6 syntax (includes arrow functions, spreading, destructuring)
+- Use Functional programming (map/filter/reduce)
+- Use React Hooks
+- Use TypeScript (and code is well typed)
+
+## Suggested Project Ideas
+
+- Picnic items
+- Card game
+- Blog
+- Forum
+- Chat bot
+- Wiki
+- School management db
+
+You are not limited to these ideas, they are just some things
+we've seen in the past that might help you get started!
+
+## Milestones
+
+The project will be divided into a number of milestones to help
+you progress in a reasonable way.
+
+Each milestone is due before lecture on its due date. Your remaining slip days are still valid for this. If you and your partner have different number of slip days left, **the partner with the most slip days left counts.** E.g. if Megan and Ashneel are a group and Megan has 5 slip days and Ashneel has 3. We are allowed 5 slip days for the final project. **You are allowed to use at most 3 slip days per each of Milestones 1, 2, 3.**
+
+### Team Matching Form
+
+**Due: November 7 by 11:59 PM (no slip days)**
+
+Please fill out [**this team matching form**](https://forms.gle/kzqW3r5fdP1u5Yix9) by **Saturday, Nov. 7th**. This is a hard deadline because I want to pair you up on Sunday so you have enough time to work on Milestone 0 (due by next class).
+
+### Milestone 0
+
+**Due: November 10 by 3:59 PM (no slip days)**
+
+Upload to CMS a mock and description of your web app! Your mock can be a picture (handdrawn or typed) and the description should be in `pdf` form. Your mock should include sketches of the different pages or views in your application. Your description should be at least 200 words and elaborate on what your project will be about. Also consider how you will structure your database schema! (What tables will you need, how will they be related, and what will they store). Please zip both of these together to submit to CMS.
+
+### Milestone 1
+
+**Due: December 1 by 3:59 PM**
+
+Get started implementing either the frontend or backend (you choose).
+If you're doing frontend, you can use hardcoded data for now just to get the layout done.
```

Comment:
May sound somewhat silly but perhaps an example/elaboration is required on what "hardcoded data" is in case some are not familiar